shell/src/main/java/alluxio/cli/validation/HdfsImpersonationValidationTask.java

```diff
@@ -0,0 +1,264 @@
+              alluxioKey.toString(), hdfsKey));
+    }
+    return new ValidateUtils.TaskResult(state, taskName, msg.toString(), advice.toString());
+  }
+
+  // At least the current host should be in hadoop.proxy.<username>.hosts
+  private ValidateUtils.TaskResult validateImpersonationHosts() {
+    String taskName = "Validate proxyuser hosts";
+    ValidateUtils.State state = ValidateUtils.State.OK;
+    StringBuilder msg = new StringBuilder();
+    StringBuilder advice = new StringBuilder();
+
+    // current host
+    String localhost = NetworkAddressUtils.getLocalHostName(1000);
+
+    // All the possible users Alluxio needs
+    Set<String> allUsers = Sets.union(mImpersonationUsers.keySet(), mImpersonationGroups.keySet());
+    for (String userName : allUsers) {
+      msg.append(String.format("User %s is configured to allow impersonation is Alluxio. ",
+              userName));
+
+      String hdfsKey = String.format("hadoop.proxyuser.%s.hosts", userName);
+      if (!mCoreConf.containsKey(hdfsKey)) {
+        state = ValidateUtils.State.FAILED;
+        msg.append("But the user is not allowed to use impersonation on this host.");
+        advice.append(String.format("Please configure %s to contain %s. ", hdfsKey, localhost));
+        continue;
+      }
+
+      String proxyHosts = mCoreConf.get(hdfsKey);
+      // Consider wildcard separately
+      if (proxyHosts.equals(ImpersonationAuthenticator.WILDCARD)) {
+        msg.append(String.format("The user is enabled for impersonation from all hosts. "));
+        continue;
+      }
+      // If wildcard is not used and the localhost is not in the permitted list
+      if (!proxyHosts.contains(localhost)) {
+        state = ValidateUtils.State.FAILED;
+        msg.append(String.format("But %s does not contain host %s. ", hdfsKey, localhost));
+        advice.append(String.format("Please enable host %s in %s. ", localhost, hdfsKey));
+        continue;
+      }
+
+      // Passed all checks
+      msg.append(String.format("Host %s is enabled to use impersonation in HDFS. ", localhost));
+    }
+    return new ValidateUtils.TaskResult(state, taskName, msg.toString(), advice.toString());
+  }
+
+  @Override
+  public ValidateUtils.TaskResult validate(Map<String, String> optionMap) {
+    if (shouldSkip()) {
+      return new ValidateUtils.TaskResult(ValidateUtils.State.SKIPPED, getName(),
+              mMsg.toString(), mAdvice.toString());
+    }
+
+    ValidateUtils.TaskResult loadConfig = loadHdfsConfig();
+    if (loadConfig.getState() != ValidateUtils.State.OK) {
+      return loadConfig;
+    }
+
+    // TODO(jiacheng): do we want to check the current user even if
+    //  there's no impersonation setting?
+    switch (mMode) {
+      case USERS:
+        return validateImpersonationUsers();
+      case GROUPS:
+        return validateImpersonationGroups();
+      case HOSTS:
+        return validateImpersonationHosts();
```

Comment:
One limitation of the current check is, we can only know which users need impersonation based on the Alluxio properties. If the user does not define `alluxio.security.impersonation.bob.users` field, there's no way to know if bob needs to be validated.

Reply:
@LuQQiu One slightly better way is to define a list of users that we "think" should have impersonation, like "presto, yarn, spark", and give a warning if they are not configured. But then the drawback is, we will subsequently complain if those users are not configured in hadoop core-site.xml, which may be confusing to those who don't need these users.